### PR TITLE
fix(listbox): add reset function

### PR DESCRIPTION
**Pull request by gerjanvangeest**
## What I did

1. Add reset function to listbox
  - improved version of: https://github.com/ing-bank/lion/pull/710

### CODE REVIEW

File: packages/listbox/test-suites/ListboxMixin.suite.js [hunk 1]
```diff
@@ -383,6 +383,52 @@ export function runListboxMixinSuite(customConfig = {}) {
...
+        `);
+        el.setCheckedIndex(1);
+        expect(el.modelValue).to.equal(20);
+        el.reset();
+        expect(el.modelValue).to.equal('');
+      });
+
+      it('should reset selected value to initial values when multiple-choice', async () => {
+        const el = await fixture(html`
+          <${tag} multiple-choice has-no-default-selected>
```
Comment (gerjanvangeest) [inline packages/listbox/test-suites/ListboxMixin.suite.js:418]: Not fully happy to have to add `has-no-default-selected`, but that is not part of this pull request

File: packages/listbox/src/ListboxMixin.js [hunk 2]
```diff
@@ -252,6 +252,7 @@ const ListboxMixinImplementation = superclass =>
       super.firstUpdated(changedProperties);
 
       this.__moveOptionsToListboxNode();
+      this._initialCheckedIndex = this.checkedIndex;
```
Comment (tlouisse) [inline packages/listbox/src/ListboxMixin.js:255]: Why not save modelValue instead (we do it in other places as wel, since this is SSOT and all other values are derived from it)?

File: .changeset/slimy-windows-sing.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@lion/listbox': patch
```
Comment (tlouisse) [inline .changeset/slimy-windows-sing.md:2]: maybe a minor even

File: packages/listbox/src/ListboxMixin.js (hunk 2 above)
Comment (gerjanvangeest) [reply packages/listbox/src/ListboxMixin.js:255]: Since we loop through all child form elements to reset the old checked state, which goes the easiest with `index`

Comment (gerjanvangeest) [reply packages/listbox/src/ListboxMixin.js:255]: But using the `modelValue` makes all other code way easier. So i'll update it

File: packages/listbox/src/ListboxMixin.js [hunk 4]
```diff
@@ -252,6 +252,8 @@ const ListboxMixinImplementation = superclass =>
       super.firstUpdated(changedProperties);
 
       this.__moveOptionsToListboxNode();
+      /** @type {any[]} */
+      this._initialModelValue = this.modelValue;
```
Comment (daKmoR) [inline packages/listbox/src/ListboxMixin.js:256]: I am wondering if we should better use
```js
this.registrationComplete.then(() => {
  this._initialModelValue = this.modelValue;
});
```

to be more "safe" if some child maybe takes slightly longer 😅

File: packages/listbox/src/ListboxMixin.js [hunk 5]
```diff
@@ -277,6 +281,20 @@ const ListboxMixinImplementation = superclass =>
...
 
+    resetInteractionState() {
+      super.resetInteractionState();
+      this.submitted = false;
+    }
+
+    reset() {
+      this.formElements.forEach(formElement => {
+        // eslint-disable-next-line no-param-reassign
+        formElement.checked = formElement.choiceValue === this._initialModelValue;
```
Comment (daKmoR) [inline packages/listbox/src/ListboxMixin.js:292]: would that not put a `checked` property an every form element? even normal inputs? 🤔

Comment (gerjanvangeest) [reply packages/listbox/src/ListboxMixin.js]: Normal inputs can't be a child of the listbox (ChoiceGroupMixin will throw an error).
Checkbox and radio can be placed inside the listbox, but this breaks the whole listbox, but i don't think we have to guard this, and especially not within the reset function.

File: packages/form-integrations/docs/15-features-overview.md [hunk 6]
```diff
@@ -94,12 +96,23 @@ export const main = () => {
           <lion-radio .choiceValue=${'brontosaurus'} label="brontosaurus"></lion-radio>
           <lion-radio .choiceValue=${'diplodocus'} label="diplodocus"></lion-radio>
         </lion-radio-group>
+        <lion-listbox name="favoriteFruit" label="Favorite fruit">
```
Comment (tlouisse) [inline packages/form-integrations/docs/15-features-overview.md:99]: 👍

File: packages/listbox/src/ListboxMixin.js [hunk 7]
```diff
@@ -381,6 +385,21 @@ const ListboxMixinImplementation = superclass =>
       /* eslint-enable no-param-reassign */
     }
 
+    resetInteractionState() {
+      super.resetInteractionState();
+      this.submitted = false;
+    }
+
+    reset() {
+      this.formElements.forEach(formElement => {
```
Comment (tlouisse) [inline packages/listbox/src/ListboxMixin.js:394]: It should be possible to do: `this.modelValue = this._initialModelValue` (since it's SSOT and will automatically sync checked state under the hood)

Comment (gerjanvangeest) [reply packages/listbox/src/ListboxMixin.js]: Yes indeed, although it required a small fix under that hood :)

**Review by tlouisse — APPROVED** [review]
